SlowCheetah.JDT/JsonTransformation.cs

```diff
@@ -79,7 +79,7 @@ public JsonTransformation(Stream transform, IJsonTransformationLogger logger)
         /// <param name="sourceFile">The path to the json file to be transformed</param>
         /// <param name="result">The stream to write the result into</param>
         /// <returns>True if the transformations were completed</returns>
-        public bool Apply(string sourceFile, out Stream result)
+        public bool TryApply(string sourceFile, out Stream result)
```

Comment:
We need to rethink this API. I didn't realize in my previous comment the logger was part of the constructor. `TryApply` will currently throw when the user has no logger supplied, which is not what a `Try` function should do. I think we need an `Apply` overload that propagates the exception regardless of logger being provided. Then the `TryApply` never throws logger or not.